watertap/unit_models/Bipolar_Electrodialysis_0D.py

```diff
@@ -0,0 +1,2343 @@
+            doc="Linear velocity of flow in the acid channel of the bipolar membrane",
+        )
+        # Parameters for bipolar membrane operation
+        self.elec_field_non_dim = Var(
+            self.flowsheet().time,
+            initialize=1,
+            # bounds=(0, 1e32),
+            units=pyunits.dimensionless,
+            doc="Limiting current density across the bipolar membrane as a function of the normalized length",
+        )
+        self.relative_permittivity = Var(
+            self.membrane_set,
+            initialize=30,
+            bounds=(1, 80),
+            domain=NonNegativeReals,
+            units=pyunits.dimensionless,
+            doc="Relative permittivity",
+        )
+        self.membrane_fixed_charge = Var(
+            self.membrane_set,
+            initialize=1.5e3,
+            bounds=(1e-1, 1e5),
+            units=pyunits.mole * pyunits.meter**-3,
+            doc="Membrane fixed charge",
+        )
+        self.kr = Var(
+            self.membrane_set,
+            initialize=1.33 * 10**11,
+            bounds=(1e-6, 1e16),
+            units=pyunits.L * pyunits.mole**-1 * pyunits.second**-1,
+            doc="Re-association rate constant",
+        )
+        self.k2_zero = Var(
+            self.membrane_set,
+            initialize=2 * 10**-5,
+            bounds=(1e-10, 1e2),
+            units=pyunits.second**-1,
+            doc="Dissociation rate constant at no electric field",
+        )
+        self.salt_conc_aem = Var(
+            self.membrane_set,
+            initialize=1e3,
+            bounds=(1e-8, 1e6),
+            units=pyunits.mole * pyunits.meter**-3,
+            doc="Salt concentration on the base channel of the bipolar membrane",
+        )
+        self.salt_conc_cem = Var(
+            self.membrane_set,
+            initialize=1e3,
+            bounds=(1e-6, 1e4),
+            units=pyunits.mole * pyunits.meter**-3,
+            doc="Salt concentration on the acid channel of the bipolar membrane",
+        )
+        self.diffus_mass = Var(
+            # self.membrane_set,
+            initialize=2e-9,
+            bounds=(1e-16, 1e-6),
+            units=pyunits.meter**2 * pyunits.second**-1,
+            doc="The mass diffusivity of the solute as molecules (not individual ions)",
+        )
+        self.conc_water = Var(
+            self.membrane_set,
+            initialize=55 * 1e3,
+            bounds=(1e-2, 1e6),
+            units=pyunits.mole * pyunits.meter**-3,
+            doc="Concentration of water within the channel",
+        )
+
+        # Limiting operation quantity
+        self.current_dens_lim_bpem = Var(
+            self.flowsheet().time,
+            initialize=1e2,
+            bounds=(0, 1e5),
+            units=pyunits.amp * pyunits.meter**-2,
+            doc="Limiting current density across the bipolar membrane",
+        )
+
+        # Fluxes Vars for constructing mass transfer terms
+        self.generation_cem_flux_in = Var(
```

Comment:
These appear to be additive components of the flux equations, are the intermediate variables just for tractability?